### PR TITLE
Adds New contributing guidelines from TGstation.

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Adds some new contributing guidelines from TGstation, most of them we already enforce, just makes this explicit and known to new coders.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -126,12 +140,48 @@ This wouldn't preserve the return value of the parent proc, instead, you should
 `.` is the current return value of the current proc.
 If you have the parent call at the end of the proc, use `return ..()` instead, it's slightly faster and more readable.
 
+### All `process` procs need to make use of delta-time and be frame independent
```
Comment (SplinterGP) [inline .github/CONTRIBUTING.md:143]: i don't know if we actually have delta time implemented

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -319,12 +463,225 @@ This prevents nesting levels from getting deeper then they need to be.
...
+
+* The ttl for data from the database is 10 seconds. You must have a compelling reason to store and reuse data for longer then this.
+
+* Do not write stored and transformed data to the database, instead, apply the transformation to the data in the database directly.
+	* ie: SELECTing a number from the database, doubling it, then updating the database with the doubled number. If the data in the database changed between step 1 and 3, you'll get an incorrect result. Instead, directly double it in the update query. `UPDATE table SET num = num*2` instead of `UPDATE table SET num = [num]`.
+	* if the transformation is user provided (such as allowing a user to edit a string), you should confirm the value being updated did not change in the database in the intervening time before writing the new user provided data by checking the old value with the current value in the database, and if it has changed, allow the user to decide what to do next.
```
Comment (SplinterGP) [inline .github/CONTRIBUTING.md:472]: i'm not sure about these SQL stuff.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -319,12 +463,225 @@ This prevents nesting levels from getting deeper then they need to be.
...
+for (var/i in 1 to 12)
+
+// Better
+for (var/month in 1 to 12)
+
+// Bad, only use `i` for numeral loops
+for (var/i in reagents)
+```
+
+### Icons are for image manipulation and defining an obj's `.icon` var, appearances are for everything else.
```
Comment (SplinterGP) [inline .github/CONTRIBUTING.md:588]: we still use images around code mostly but this should be the ideal as appearances(/mutable_appearances) should be preferred as they are more performant when changing many appearance related vars.

**Review by spookydonut — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (BraveMole) [reply .github/CONTRIBUTING.md:143]: We don't

**Pull request merged**